src/aims_ui/page_singlesearch.py

```diff
@@ -0,0 +1,48 @@
+import os
+from flask import render_template, request, session
+from flask_login import login_required
+from . import app
+from .cookie_utils import save_input, load_input, get_all_inputs, delete_input, load_save_store_inputs
+from .api_interaction import api
+from .models.get_endpoints import get_endpoints
+from .models.get_fields import get_fields
+from .models.get_addresses import get_addresses
+import json
+
+
+page_name =  'singlesearch'
+
+@login_required
+@app.route(f'/{page_name}', methods=['GET', 'POST'])
+def singlesearch():
+
+  if request.method == 'GET':
+    delete_input(session)
+    return render_template(
+        f'{page_name}.html',
+        searchable_fields=get_fields(page_name),
+        endpoints=get_endpoints(called_from=page_name),
+    )
+
+  searchable_fields = get_fields(page_name)
+  all_user_input = load_save_store_inputs(searchable_fields,request,session,)
+
+
+  result = api(
+      '/addresses',
+      page_name,
+      all_user_input,
+  )
+
+  if result.status_code == 200:
+    matched_addresses = get_addresses(result.json(),page_name)
+  else:
+    matched_addresses = ''
```

Comment:
In this and other pages, it might be good to either raise an error, or render a template that can display the error, or the real return code and the reason for the failure will be lost.

Reply:
Have added an error page which a user is routed to with the response code. This could be potentially added to later using swagger to specify any errors more specifically